api/src/main/java/org/datacleaner/connection/DatastoreDescriptor.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.connection;
+
+import java.util.List;
+
+/**
+ * A descriptor class for a datastore types in DataCleaner. 
+ *
+ */
+public interface DatastoreDescriptor {
+    
+    public String getName();
+    
+    public String getDescription();
+    
+    public Class<? extends Datastore> getDatastoreClass();
+    
+    public boolean isUpdatable();
+    
+    public List<String> getTags();
```

Comment:
From what I can see this is used to depict just if the datastore is put under the "Cloud service" or "Database" button in the dropzone screen. But a concept like tags is very generic and hard to interpret from reading this interface. And since it's entirely a UI representation thing, maybe having this categorization just in the UI would be better.